platform/backend/core/src/main/kotlin/io/hamal/core/adapter/Func.kt

```diff
@@ -112,8 +112,9 @@ class FuncAdapter(
         versionToDeploy: CodeVersion,
         responseHandler: (FuncDeploySubmitted) -> T
     ): T {
-        ensureFuncExists(funcId)// FIXME we dont have to check it here - its already tested in submitRequst()
-        return responseHandler(submitRequest(funcId, versionToDeploy)) // FIXME-53
+        val func = funcQueryRepository.get(funcId)
+        codeQueryRepository.get(func.code.id, versionToDeploy)
+        return responseHandler(submitRequest(funcId, versionToDeploy))
```

Comment:
just do a passthrough here - because calling a database always costs and you do it twice